tests/array.rs

```diff
@@ -2819,3 +2819,11 @@ fn test_split_complex_invert_axis()
     assert_eq!(cmplx.re, a.mapv(|z| z.re));
     assert_eq!(cmplx.im, a.mapv(|z| z.im));
 }
+
+#[test]
+fn test_slice_assing()
+{
+    let mut a = array![0, 1, 2, 3, 4];
+    *a.slice_mut(s![1..3]) += 1;
```

Comment:
Wasn't this forbidden before?

Reply:
Impossible, in fact! That's one of the big motivators for the change. `slice_mut` provides an `ArrayViewMut`, which couldn't be dereferenced. Now that `ArrayRef` exists, this operation is valid.